src/api/providers/aistupidlevel.ts

```diff
@@ -0,0 +1,98 @@
+import { Anthropic } from "@anthropic-ai/sdk"
+import OpenAI from "openai"
+
+import { aiStupidLevelDefaultModelId, aiStupidLevelDefaultModelInfo } from "@roo-code/types"
+
+import { ApiHandlerOptions } from "../../shared/api"
+
+import { ApiStream } from "../transform/stream"
+import { convertToOpenAiMessages } from "../transform/openai-format"
+
+import type { SingleCompletionHandler, ApiHandlerCreateMessageMetadata } from "../index"
+import { RouterProvider } from "./router-provider"
+
+const AISTUPIDLEVEL_DEFAULT_TEMPERATURE = 0.7
+
+export class AIStupidLevelHandler extends RouterProvider implements SingleCompletionHandler {
+	constructor(options: ApiHandlerOptions) {
+		super({
+			options,
+			name: "aistupidlevel",
+			baseURL: "https://api.aistupidlevel.info/v1",
+			apiKey: options.aiStupidLevelApiKey,
+			modelId: options.aiStupidLevelModelId,
+			defaultModelId: aiStupidLevelDefaultModelId,
+			defaultModelInfo: aiStupidLevelDefaultModelInfo,
+		})
+	}
+
+	override async *createMessage(
```

Comment:
There's no non-streaming path implemented here.   But your implementation of completePrompt() below certainly implies your service supports it.

Is it intentional to only support streaming for agentic activity?